### DISCUSSION TITLE
Accessing resource attributes as done in Terraform

### DISCUSSION THREAD

**Discussion by denzhel** [crossplane/crossplane#5047] [Q&A]
I'd like to create an ECS cluster with `aws-provider-ecs` using three managed resources:
* Cluster
* TaskDefinition
* Service

I'm creating a helm-chart that wraps this to enable our developers to create their own clusters.
The thing is that in order to create a managed resourced of kind `Service` I need the ARN of both Cluster and TaskDefinition.
While at Terraform I'd use:
```
cluster         = aws_ecs_cluster.integrations_inbound.id
task_definition = aws_ecs_task_definition.alert_persistence.arn
```

How do I do that with Crossplane/ Helm ? 

Thank you

**Comment by jbw976** [comment]
This **sounds** like a typical use case of Composition in Crossplane 🤓 
https://docs.crossplane.io/latest/concepts/compositions/

If you create those resources together within a Composition, you can reference the fields/values/status from each other.  Check out this example of creating a NodeGroup along with an EKS cluster - the node group is pulling a bunch of values from other resources, e.g. the cluster, subnets, and role ARN:

https://github.com/upbound/configuration-aws-eks/blob/main/apis/composition.yaml#L303-L311

This may be easier to manage using Crossplane's composition model instead of wrapping managed resources directly with Helm.
